rcl/CMakeLists.txt

```diff
@@ -47,10 +47,17 @@ macro(target)
     ARCHIVE DESTINATION lib
     LIBRARY DESTINATION lib
     RUNTIME DESTINATION bin)
+
+  # rcl_lib_dir is passed as APPEND_LIBRARY_DIRS for each ament_add_gtest call so
+  # the librcl that they link against is on the library path.
+  # This is especially important on Windows.
+  # This is overwritten each loop, but which one it points to doesn't really matter.
+  set(rcl_lib_dir $<TARGET_FILE_DIR:${PROJECT_NAME}${target_suffix}>)
```

Comment:
You should quote the value since it might contain spaces and shouldn't become a list.

And every usage below should also have quotes around the variable.